eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -54,67 +54,74 @@
-            VariantWithSamplesAndAnnotation variant = new VariantWithSamplesAndAnnotation(variantEntity.getChromosome(),
+            /*VariantWithSamplesAndAnnotation variant = new VariantWithSamplesAndAnnotation(variantEntity.getChromosome(),
                     variantEntity.getStart(), variantEntity.getEnd(), variantEntity.getReference(),
                     variantEntity.getReference(), variantEntity.getMainId());
-
-            variantEntity.getIds().forEach(id -> variant.addId(id));
-            rootVariantEntities.add(variant);
+            variantEntity.getIds().forEach(id -> variant.addId(id));*/
+            //rootVariantEntities.add(variant);
+            variantCoreInfoList.add(new VariantCoreInfo(variantEntity.getChromosome(),variantEntity.getStart(),
+                    variantEntity.getEnd(),variantEntity.getReference(),variantEntity.getAlternate(),
+                    variantEntity.getIds(),variantEntity.getType(),variantEntity.getLength(),variantEntity.getHgvs()));
         });
 
-        QueryResult<VariantWithSamplesAndAnnotation> queryResult = buildQueryResult(rootVariantEntities,
+        QueryResult<VariantCoreInfo> queryResult = buildQueryResult(variantCoreInfoList,
                 numTotalResults);
 
-        Link annotationLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getAnnotations(variantId, species,
-                annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(), "annotation");
+        Link annotationLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getAnnotations(variantCoreString,
+                species, assembly, annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(),
+                "annotation");
 
-        Link sourceEntriesLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getSourceEntries(variantId, species,
-                annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(), "sourceEntries");
+        Link sourcesLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getsources(variantCoreString, species,
+                assembly, annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(), "sources");
 
         List<Link> links = new ArrayList<>();
-        links.add(sourceEntriesLink);
+        links.add(sourcesLink);
         links.add(annotationLink);
 
-        QueryResponse<QueryResult<VariantSourceEntryWithSampleNames>> sourceEntries = getSourceEntries(variantId,
-                species, annotationVepVersion, annotationVepCacheVersion, response);
-
         return new Resource<>(setQueryResponse(queryResult), links);
     }
 
-    private String checkErrorHelper(String variantId, String annotationVepVersion, String annotationVepCacheVersion,
-                                    String species) {
-        if (!variantId.contains(":")) {
+    private String checkParameters(String variantCoreString, String annotationVepVersion,
```

Comment:
Part of this method throws `IllegalArgumentException` and another part returns the error message. Please make all branches throw an exception.

Reply:
Fixed :)